source/opt/inline_pass.cpp

```diff
@@ -270,30 +319,120 @@ std::unique_ptr<BasicBlock> InlinePass::AddGuardBlock(
+      break;
+    }
+    chain_iter_id = GetInlinedOperand(new_inlined_at_in_chain);
+  } while (chain_iter_id);
+
+  // Put |inlined_to_caller| into the end of the chain.
+  if (last_inlined_at_in_chain != nullptr) {
+    if (HasInlinedOperand(last_inlined_at_in_chain)) {
+      UpdateInlinedOperand(last_inlined_at_in_chain, inlined_to_caller);
+    } else {
+      last_inlined_at_in_chain->AddOperand(
+          {SPV_OPERAND_TYPE_RESULT_ID, {inlined_to_caller}});
+    }
+  }
+
+  // Keep the new chain information that will be reused it.
+  inlined_at_ctx.UpdateInlinedAtContext(callee_inlined_at, chain_head_id);
+  return chain_head_id;
+}
+
 InstructionList::iterator InlinePass::AddStoresForVariableInitializers(
     const std::unordered_map<uint32_t, uint32_t>& callee2caller,
+    DebugInlinedAtContext& inlined_at_ctx,
     std::unique_ptr<BasicBlock>* new_blk_ptr,
     UptrVectorIterator<BasicBlock> callee_first_block_itr) {
-  auto callee_var_itr = callee_first_block_itr->begin();
-  while (callee_var_itr->opcode() == SpvOp::SpvOpVariable) {
-    if (callee_var_itr->NumInOperands() == 2) {
-      assert(callee2caller.count(callee_var_itr->result_id()) &&
+  auto callee_itr = callee_first_block_itr->begin();
+  while (callee_itr->opcode() == SpvOp::SpvOpVariable ||
+         callee_itr->GetOpenCL100DebugOpcode() ==
+             OpenCLDebugInfo100DebugDeclare) {
+    if (callee_itr->opcode() == SpvOp::SpvOpVariable &&
+        callee_itr->NumInOperands() == 2) {
+      assert(callee2caller.count(callee_itr->result_id()) &&
              "Expected the variable to have already been mapped.");
-      uint32_t new_var_id = callee2caller.at(callee_var_itr->result_id());
+      uint32_t new_var_id = callee2caller.at(callee_itr->result_id());
 
       // The initializer must be a constant or global value.  No mapped
       // should be used.
-      uint32_t val_id = callee_var_itr->GetSingleWordInOperand(1);
-      AddStore(new_var_id, val_id, new_blk_ptr);
+      uint32_t val_id = callee_itr->GetSingleWordInOperand(1);
+      AddStore(new_var_id, val_id, new_blk_ptr, callee_itr->dbg_line_inst(),
+               BuildDebugScope(callee_itr->GetDebugScope(), inlined_at_ctx));
     }
-    ++callee_var_itr;
+    if (callee_itr->GetOpenCL100DebugOpcode() ==
+        OpenCLDebugInfo100DebugDeclare) {
+      InlineSingleInstruction(
+          callee2caller, new_blk_ptr->get(), &*callee_itr,
+          BuildDebugInlinedAtChain(callee_itr->GetDebugScope().GetInlinedAt(),
+                                   inlined_at_ctx),
+          true);
+    }
+    ++callee_itr;
   }
-  return callee_var_itr;
+  return callee_itr;
 }
 
-bool InlinePass::InlineInstructionInBB(
+bool InlinePass::InlineSingleInstruction(
     const std::unordered_map<uint32_t, uint32_t>& callee2caller,
-    BasicBlock* new_blk_ptr, const Instruction* inst) {
+    BasicBlock* new_blk_ptr, const Instruction* inst, uint32_t dbg_inlined_at,
+    bool use_new_id) {
```

Comment:
I don't get why you need `use_new_id`.  Is there a reason the result ids of the debug instruction should not be mapped?  Is there a test that points out the reason for this?

Reply:
Ah .. I was confused. I reverted it.